core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -150,9 +150,20 @@ TileSizeProperties get_tile_size_properties(const ExecutionSpace&) {
 
 // multi-dimensional iteration pattern
 template <typename... Properties>
-struct MDRangePolicy : public Kokkos::Impl::PolicyTraits<Properties...> {
-  using traits       = Kokkos::Impl::PolicyTraits<Properties...>;
-  using range_policy = RangePolicy<Properties...>;
+struct MDRangePolicy;
+
+// Note: If MDRangePolicy has a primary template, implicit CTAD (deduction
+// guides) are generated -> MDRangePolicy<> by some compilers, which is
```

Comment:
just curious, which compilers do this?

Reply:
Implicit guides are generated from the primary template.

Because there are no template parameters to deduce in a parameter pack, both clang and gcc deduce a matching ctor to `MDRangePolicy<>`, while nvcc will generate an error.  Unfortunately, I find the standard wording on implicit guides to be fairly inscrutable (and cppreference doesn't help, as all it does is quote the standard) :-(.  While I'd certainly like to know, practically speaking, it doesn't matter to us which one is correct, as we have to support all those compilers anyway.

Since MDRangePolicy always requires at least one template parameter (`Rank`), any guides that deduce to `MDRangePolicy<>` are just wrong.